lnwallet/commitment.go

```diff
@@ -122,6 +122,12 @@ func DeriveCommitmentKeys(commitPoint *btcec.PublicKey,
 	// we'll use that directly, and ignore the commitPoint tweak.
 	if tweaklessCommit {
 		keyRing.RemoteKey = remoteBasePoint
+
+		// If this is a tweakless remote commitment, then we can safely
+		// blank out the SingleTweak value as it isn't needed.
+		if !isOurCommit {
+			keyRing.LocalCommitKeyTweak = nil
+		}
```

Comment:
I get a bit confused with all the local and remote perspectives, but isn't it the case now that both local and remote keys are untweaked?

Reply:
I did it this way to not change current behavior, where we only nil it for our own. But you are right, also from the remote perspective it could/should be tweakless, it just wasn't needed since we never use this variable from that perspective.
Changed: https://github.com/lightningnetwork/lnd/pull/3829/commits/38c11a43297830c443d69b6f65c6274da9aae2aa

Reply:
Reverted back to the original change here, since it the `to_local` key stays tweaked even for tweakless commits.

Reply:
We can safely blank it out, but why would we? Does this protect us against something?

Reply:
The comment was just copied, and yes a bit confusing. Updated: https://github.com/lightningnetwork/lnd/pull/3829/commits/0e46272440be020f82e107e7698ac5265595e443